mne/preprocessing/maxwell.py

```diff
@@ -505,3 +539,38 @@ def _update_sss_info(raw, origin, int_order, ext_order, nsens):
         raw.info['proc_history'] = [proc_block]
 
     return raw
+
+
+def _overlap_projector(data_int, data_res, delta=0.02):
+    """Calculate projector for removal of subspace intersection in tSSS"""
+    # delta necessary to deal with noise when finding identical signal
+    # directions in the subspace. See the end of the Results section in [2]_
+
+    # Note that the procedure here is an updated version of [2]_ using
+    # residuals instead of internal/external spaces directly. This provides
+    # data with higher rank to work with.
+
+    # Normalize data
+    normed_data_int = data_int / norm(data_int)
+    normed_data_res = data_res / norm(data_res)
+
+    # Compute orth to get temporal bases. Matrices must have shape
+    # (n_samps x effective_rank) when passed into svd computation
+    orth_int = orth(normed_data_int.T)
+    orth_res = orth(normed_data_res.T)
```

Comment:
That's pretty terrible, we need to optimize that. Any idea how long it takes in MATLAB? Can you try it in `single` precision to see if it gets much speedup? It's possible there is some way to make it faster. Sometimes for the algorithms in `scipy` they have to choose what to use under the hood, this might be a case where we need to figure out the best way to do it for these data.